osclib/stagingapi.py

```diff
@@ -1072,25 +1072,27 @@ def map_ring_package_to_subject(self, project, pkg):
 
         return project
 
-    def get_sub_packages(self, pkg, project=None):
+    def get_sub_packages(self, package, project=None):
         """
         Returns a list of packages that need to be linked into rings
         too. A package is actually a tuple of project and package name
         """
         ret = []
         if not project:
-            project = self.ring_packages.get(pkg)
+            project = self.ring_packages.get(package)
         if not project:
             return ret
-        url = self.makeurl(['source', project, pkg],
+        url = self.makeurl(['source', project, package],
                            {'cmd': 'showlinked'})
 
         # showlinked is a POST for rather bizzare reasons
         f = http_POST(url)
         root = ET.parse(f).getroot()
 
         for pkg in root.findall('package'):
-            ret.append((pkg.get('project'), pkg.get('name')))
+            # ensure sub-package is valid
+            if pkg.get('project') in self.rings and pkg.get('name') != package:
```

Comment:
do you planning to change `get_sub_package` to cope with adi? because `get_sub_package` is mainly for ring package, if you've plan to do that then yes this can be a problem against adi, but if do so, there is another problem, `showlinked` call does not make sense for adi package, the **layout**(all specs are identical to the new request; all sub-package's link are created in the devel project) is not easy to guaranteed. For https://github.com/openSUSE/osc-plugin-factory/pull/948 , created main package in adi and then create sub-packages for rest of specfile is the proper way IMO.